python/cudf_polars/cudf_polars/dsl/traversal.py

```diff
@@ -49,6 +49,40 @@ def traversal(nodes: Sequence[NodeT]) -> Generator[NodeT, None, None]:
                 lifo.append(child)
 
 
+def post_traversal(nodes: Sequence[NodeT]) -> Generator[NodeT, None, None]:
+    """
+    Post-order traversal of nodes in an expression.
+
+    Parameters
+    ----------
+    nodes
+        Roots of expressions to traverse.
+
+    Yields
+    ------
+    Unique nodes in the expressions, child before parent, children
+    in-order from left to right.
+    """
+    seen = set()
+    lifo = []
```

Comment:
If this ends up collecting a lot of nodes, could use collections.deque for ~O(1) appends/pops

Reply:
Okay, thanks. I switched to deque - The perf difference was small for the random TPC-H queries I checked (~0.2ms vs ~0.4ms for traversal), but there was no penalty at small scale (so why not?)